### ISSUE TITLE
println!() breaks incr. comp.

### ISSUE THREAD

**Issue by michaelwoerister** [rust-lang/rust#35593]
The following test, when added to the `incremental` test suite, fails:
```
// revisions:rpass1 rpass2

#![feature(rustc_attrs)]
#![rustc_partition_reused(module="println", cfg="rpass2")]

fn main() {
    println!("hello world");
}
```
Something causes the hash for `main` to be different in the two revisions.

**Comment by michaelwoerister** [comment]
cc @nikomatsakis

**Comment by nikomatsakis** [comment]
This is caused by the fact that typeck rewrites the result of name resolution in place, and hence the name resolution hashes after typeck are different than the ones from before. =(

The easiest fix is probably #35549
